File: app/src/main/java/com/example/alarmmanager/MainActivity.java
package com.example.alarmmanager;

import androidx.appcompat.app.AppCompatActivity;

import android.app.AlarmManager;
import android.app.PendingIntent;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {
    Button startBtn,stopBtn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        startBtn=findViewById(R.id.start);
        stopBtn=findViewById(R.id.stop);

        stopBtn.setOnClickListener(this);
        startBtn.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        if(v.getId()==R.id.start)
        {
            Log.v("Exe","start");
            Intent intent=new Intent(MainActivity.this,MyAlarmService.class);
            PendingIntent pendingIntent= PendingIntent.getBroadcast(MainActivity.this,0,intent,PendingIntent.FLAG_UPDATE_CURRENT);
            AlarmManager alarmManager = (AlarmManager) getSystemService(ALARM_SERVICE);

            long timeInMillis = System.currentTimeMillis();
            long intervalInMillis = 5000;
            alarmManager.setRepeating(AlarmManager.RTC_WAKEUP, timeInMillis, intervalInMillis, pendingIntent);
        }
        if (v.getId()==R.id.stop)
        {
            Intent intent = new Intent(MainActivity.this, MyAlarmService.class);
            PendingIntent pendingIntent = PendingIntent.getBroadcast(MainActivity.this, 0, intent, PendingIntent.FLAG_UPDATE_CURRENT);
            AlarmManager alarmManager = (AlarmManager) getSystemService(ALARM_SERVICE);
            alarmManager.cancel(pendingIntent);

        }

    }
}

File: app/src/main/java/com/example/jetpackexample/MainActivity.kt
package com.example.jetpackexample

import android.annotation.SuppressLint
import android.content.Context
import android.os.Bundle
import android.view.View
import android.widget.LinearLayout
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.jetpackexample.Adapter.MyAdapter
import com.example.jetpackexample.Database.MyDatabase
import com.example.jetpackexample.ViewModels.LiveDataViewModel
import com.google.android.material.floatingactionbutton.FloatingActionButton


class MainActivity : AppCompatActivity(), View.OnClickListener {
    lateinit var floatButton: FloatingActionButton
    lateinit var viewModel: LiveDataViewModel
    lateinit var manager: FragmentManager
    lateinit var recyclerView: RecyclerView

    /*declaring the companion object*/
    companion object {
        public lateinit var roomDatabase: MyDatabase
        var instance: MainActivity? = null
        fun applicationContext(): Context {
            return instance!!.applicationContext
        }
    }


    @SuppressLint("WrongConstant")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        floatButton = findViewById(R.id.floatingActionButton)
        recyclerView = findViewById(R.id.recyclerView)

        //creating instance of database
        roomDatabase = MyDatabase.getInstance(this)!!
        manager = supportFragmentManager

        /*creating view model for the live data*/
        viewModel = ViewModelProviders.of(this).get(LiveDataViewModel::class.java)
        /*getting the value of the list*/
        viewModel.getUserData().observe(this, Observer<List<DataItems>> {
            val adapter = MyAdapter(it)
            recyclerView.layoutManager = LinearLayoutManager(instance, LinearLayout.VERTICAL, false)
            recyclerView.adapter = adapter
        })



        floatButton.setOnClickListener(this)
    }


    override fun onClick(v: View?) {
        if (v != null) {
            when (v.id) {
                R.id.floatingActionButton -> {
                    //replacing with the fragment
                    manager.beginTransaction().replace(R.id.mainLayout, MyFragment.newInstance())
                        .addToBackStack(null).commit()
                }
            }
        }
    }


    /*class MainAsynTask: AsyncTask<Void,Void,Void>()
    {
        public lateinit var arrayList: ArrayList<DataItems>

        override fun onPreExecute() {
            super.onPreExecute()
        }
        override fun doInBackground(vararg params: Void?): Void? {
            arrayList= ArrayList<DataItems>()
           var list: List<DataItems>? =roomDatabase.myDao()?.getUser()

            if (list != null) {
                for (data in  list) {
                    arrayList.add(data)
                }
            }
            return null
        }

        @SuppressLint("WrongConstant")
        override fun onPostExecute(result: Void?) {
            super.onPostExecute(result)
            val adapter=MyAdapter(arrayList)
            recyclerView.layoutManager=LinearLayoutManager(instance,LinearLayout.VERTICAL,false)
            recyclerView.adapter=adapter
        }

    }*/
}
